### ISSUE TITLE
Hard crash on STPColorUtils > perceivedBrightnessForColor

### ISSUE THREAD

**Issue by spencerkohan** [stripe/stripe-ios#951]
To make it easier to diagnose your issue, please fill out the following:

## Summary

There's a hard crash from this method if the navigation bar color is UIColor.white:

+ (CGFloat)perceivedBrightnessForColor:(UIColor *)color {
    const CGFloat *component = CGColorGetComponents(color.CGColor);
    CGFloat brightness = ((component[0] * 299) + (component[1] * 587) + (component[2] * 114)) / 1000;
    return brightness;
}

Basically if the color is defined using white/alpha rather than rgba, *component here will not have 3 values, and you get an exec bad acess

## Code to reproduce

Just try calling this method with a nav bar with the tint color set to UIColor.white

## iOS version

11

## Installation method

cocoapods

## SDK version

latest

**Comment by danj-stripe** [comment]
Hi @spencerkohan,

Thanks for finding & reporting this! I had to turn on Address Sanitization before I saw the crash, but once I did it was 100% reproducible.

I've merged a fix, and it should go out in our next release.

Thanks!
Dan
